components/dashboard/src/settings/SelectIDE.tsx

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { IDEOption, IDEOptions } from "@gitpod/gitpod-protocol/lib/ide-protocol";
+import { useContext, useEffect, useState } from "react";
+import InfoBox from "../components/InfoBox";
+import SelectableCardSolid from "../components/SelectableCardSolid";
+import Tooltip from "../components/Tooltip";
+import { getGitpodService } from "../service/service";
+import { UserContext } from "../user-context";
+import CheckBox from "../components/CheckBox";
+
+export interface SelectIDEProps {
+    showLatest?: boolean;
+}
+
+export default function SelectIDE(props: SelectIDEProps) {
+    const { showLatest = true } = props;
+
+    const { user, setUser } = useContext(UserContext);
+
+    const updateUserIDEInfo = async (selectedIde: string, useLatestVersion: boolean) => {
+        const additionalData = user?.additionalData ?? {};
+        const settings = additionalData.ideSettings ?? {};
+        settings.settingVersion = "2.0";
+        settings.defaultIde = selectedIde;
+        settings.useLatestVersion = useLatestVersion;
+        additionalData.ideSettings = settings;
+        getGitpodService()
+            .server.trackEvent({
+                event: "ide_configuration_changed",
+                properties: settings,
+            })
+            .then()
+            .catch(console.error);
+        await getGitpodService().server.updateLoggedInUser({ additionalData });
+    };
+
+    const [defaultIde, setDefaultIde] = useState<string>(user?.additionalData?.ideSettings?.defaultIde || "");
+    const actuallySetDefaultIde = async (value: string) => {
+        // force context to refresh
+        setUser({ ...user! });
```

Comment:
This `hack` was required to make the components re-render due to state change. If anyone can suggests a nicer way, happy to change it